client/index.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { render } from 'react-dom';
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import { composeWithDevTools } from 'redux-devtools-extension';
+import { BrowserRouter as Router, Route } from 'react-router-dom';
+
+import App from './components/App';
+
+import { asyncMiddleware } from './middleware';
+import rootReducer from './reducers';
+import './styles.scss';
+
+const Root = () => {
+    let middleware = composeWithDevTools(applyMiddleware(asyncMiddleware, thunk));
+    let store = createStore(
+        rootReducer, 
+        middleware
+    );
+
+    return (
+        <Provider store={store}>
+            <Router>
+                <Route path='/' component={App} />
+            </Router>
+        </Provider>
+    );
+}
+
+render(<Root />, document.getElementById("root"));
```

Comment:
This is the root of the React application. It creates the Redux store (with middleware included - `redux-thunk` and my `asyncMiddleware`). It also sets up the client side routing which is done with `react-router`. Throughout the React application you'll see `<Route />` components. These are rendered by `react-router` whenever the url matches the specified path